manifests/slave.pp

```diff
@@ -170,6 +176,16 @@
     }
   }
 
+  if $slave_args {
+    if is_array($slave_args) {
+      $_combined_slave_args = hiera_array('jenkins::slave::slave_args', $slave_args)
```

Comment:
@madelaney I figured that might be the case, but I think that makes this behavior even more confusing. Hiera already has configuration (which I *think* applies to injected class parameters) for its [merge behaviors](https://docs.puppet.com/hiera/3.2/configuring.html#mergebehavior) (also [array merge handling](https://docs.puppet.com/hiera/3.2/lookup_types.html#array-merge)).

There's also [this documentation](https://docs.puppet.com/puppet/latest/reference/lookup_quick.html#specifying-merge-behavior) which might be relevant? The myriad of merge-related documents annoys and confuses me :person_frowning: 


Anyways, I generally find accepting class parameters *plus* doing a `hiera_array` lookup of the same key to be weird.